config.go

```diff
@@ -18,6 +18,8 @@ type Spec struct {
 	Mounts []Mount `json:"mounts"`
 	// Hooks are the commands run at various lifecycle events of the container.
 	Hooks Hooks `json:"hooks"`
+	// Annotations is an unstructured key value map that may be set by external tools to store and retrieve arbitrary metadata.
+	Annotations map[string]string `json:"annotations,omitempty"`
```

Comment:
Are we sure we don't [want an opaque type here][1]?  Or will we just revisit that later when somebody needs something that doesn't fit into a `map[string]string`?

[1]: https://github.com/opencontainers/specs/pull/331#issuecomment-192477931